Avoid duplicate first or last page button, when total pages is one above truncateThreshold

edlerd opened the pull request:
## Done

- fixed pagination edge case for when the `totalPages +1 = truncateThreshold` to avoid duplicating the button for the first or last page

## QA

### QA steps

- See tests are valid and passing

webteam-app commented:
Demo starting at https://react-components-971.demos.haus

huwshimi reviewed (approved):
This looks great. It could be nice to also remove the ellipsis when on the first or last pages when all the items are displayed.

It seems to work fine if you're on the middle items:

<img width="321" alt="Screenshot 2023-09-05 at 9 38 23 am" src="https://github.com/canonical/react-components/assets/361637/4d4c59fe-bf6f-43bd-aabb-e37270f13820">

But at the start and end it displays ellipsis:

<img width="345" alt="Screenshot 2023-09-05 at 9 38 13 am" src="https://github.com/canonical/react-components/assets/361637/25449fb2-b300-4f84-80cd-5b131cb5f602">

edlerd commented:
> This looks great. It could be nice to also remove the ellipsis when on the first or last pages when all the items are displayed.

Good catch, I fixed this and added a test for this case as well.

huwshimi reviewed (approved):
Looks great! Thanks @edlerd.
